### PR TITLE
Inventory, variables, and NPC avatars

### PR DESCRIPTION
**Pull request by thequeenofspades**
Added a barebones Player class with an inventory attribute. Clicking Clickables now causes them to be added to the player's inventory. A Toast also appears at the top of the screen informing the player of what they have picked up.

An NPC avatar is now displayed on the righthand side of the screen during conversation. This avatar is changed if a different speaker enters the conversation.

Conversations can now check player variables and inventory to see if conditions are met to display responses. Actions (such as setting variables) can also be taken during a conversation.

The dialogue window now disappears (visible=false) at the end of a conversation.

Also added various new art to the assets folder and made slight updates to prologue01.json.

### CODE REVIEW

File: src/objects/DialogueWindow.js [hunk 1]
```diff
@@ -219,7 +242,9 @@ DialogueWindow.prototype.addOverflowScroll = function () {
 
 DialogueWindow.prototype.hide = function () {
   this.cleanWindow();
-  this.visible = false;
+  //this.visible = false;
+  this.dialogPanel.visible = false;
```
Comment (drcaramelsyrup) [inline src/objects/DialogueWindow.js:246]: Ah! Did this fix the old problem of the dialogue window not disappearing?

File: src/objects/ConversationManager.js [hunk 2]
```diff
@@ -38,14 +38,56 @@ ConversationManager.prototype.getCurrentText = function () {
   return this.conversation[this.idx]['text'];
 };
 
-ConversationManager.prototype.getResponses = function () {
+ConversationManager.prototype.getResponses = function (game) {
```
Comment (drcaramelsyrup) [inline src/objects/ConversationManager.js]: Should this (and getCondition) be using its own this._game object, if that's going to be our precedent going forward?

File: static/assets/conversations/prologue02.txt [hunk 3]
```diff
@@ -0,0 +1,157 @@
+:: StoryTitle
```
Comment (drcaramelsyrup) [inline static/assets/conversations/prologue02.txt]: Is this the Twine file? Are we storing it here too?

File: src/objects/ConversationManager.js [hunk 4]
```diff
@@ -38,14 +38,56 @@ ConversationManager.prototype.getCurrentText = function () {
...
+        return true;    //player does not have this variable set, or has it set to a different value
+      }
+    } else if (variable in game.player.variables && game.player.variables[variable] === value) {
+      return true;      //player has this variable set to this value
+    }
+  } else if (condition.startsWith('inv')) {
+    var item = condition.substring(3);
+    if (value.startsWith('!')) {
+      if (game.player.inventory.indexOf(item) === -1) {
+        return true;    //player does not have this inventory item
```
Comment (drcaramelsyrup) [inline src/objects/ConversationManager.js:84]: Should this be true? If the value starts with !, does that mean it's supposed to be the inverse?

**Review by drcaramelsyrup — COMMENTED** [review]
LGTM aside from a few small notes I had! Just take a looksy at the comments and see if anything needs changing, but I think I got the intentions. Otherwise you can just merge them in.

File: src/objects/DialogueWindow.js (hunk 1 above)
Comment (thequeenofspades) [reply src/objects/DialogueWindow.js:246]: Yep!

File: src/objects/ConversationManager.js (hunk 2 above)
Comment (thequeenofspades) [reply src/objects/ConversationManager.js]: Probably. I can add that in.

File: src/objects/ConversationManager.js (hunk 4 above)
Comment (thequeenofspades) [reply src/objects/ConversationManager.js:84]: That's the syntax I'm using. If it's an inventory item, adding a "!" as the prefix means that the condition passes if the player doesn't have it. If it's a variable, adding a "!" as the prefix means that the condition passes if the variable is unset, or set to a value other than whatever comes after the "!". There's also more information in the README in the Conversations folder in the Google Drive.

File: static/assets/conversations/prologue02.txt (hunk 3 above)
Comment (thequeenofspades) [reply static/assets/conversations/prologue02.txt]: Actually, that's not really necessary. I can get rid of it.

**Pull request merged**